maven-core/src/main/java/org/apache/maven/plugin/internal/DefaultPluginDependenciesResolver.java

```diff
@@ -205,6 +205,22 @@ private DependencyNode resolveInternal( Plugin plugin, Artifact pluginArtifact,
 
             node = repoSystem.collectDependencies( pluginSession, request ).getRoot();
 
+            if ( logger.isWarnEnabled() )
+            {
+                for ( DependencyNode child : node.getChildren() )
+                {
+                    if ( !child.getRelocations().isEmpty() )
+                    {
+                        org.eclipse.aether.artifact.Artifact relocated = child.getDependency().getArtifact();
+                        String message = relocated instanceof org.apache.maven.repository.internal.RelocatedArtifact
+                                ? ( ( org.apache.maven.repository.internal.RelocatedArtifact ) relocated ).getMessage()
+                                : null;
+                        logger.warn( "The artifact " + child.getRelocations().get( 0 ) + " has been relocated to "
+                                + relocated + ( message != null ? ": " + message : "" ) );
+                    }
+                }
+            }
+
```

Comment:
by definition, aren't plugin dependencies transitives (of the plugin)?
shouldn't those warnings finally be limited to current reactor projects AND only on transitive dependencies (and fully ignore plugins: too late to fix anything)?

Reply:
Good catch, I've been fooled by the not strict enough testing I added in https://github.com/apache/maven-integration-testing/pull/134/files.  It's testing the plugin's first level dependencies relocations instead of the plugin itself.